### PR TITLE
finished filtering function

### PR DESCRIPTION
**Pull request by marcinel123**
Hi Adam,

In this PR I added logic to input filed to filter through countries data.

Please have a look.
Martin

### CODE REVIEW

File: src/components/CountriesList/CountriesList.styles.tsx [hunk 1]
```diff
@@ -8,3 +8,15 @@ export const StyledDisplaySection = styled.section`
 	flex-wrap: wrap;
 	justify-content: space-around;
 `;
+
+export const StyledH4 = styled.h4`
```
Comment (AdamKniec) [inline src/components/CountriesList/CountriesList.styles.tsx]: change to p 👍

File: src/components/CountriesList/CountriesList.tsx [hunk 2]
```diff
@@ -1,23 +1,31 @@
...
 import { CountryCard } from "../countryCard/CountryCard";
 import { useCountriesContext } from "../../context/CountriesDataContext";
 import { CountriesProps } from "./CountriesProps";
 
 export const CountriesList = () => {
-	const { error, countries, selectCountryRegion } = useCountriesContext();
+	const { inputValue, error, countries, selectCountryRegion } =
+		useCountriesContext();
+
+	const filteredData = countries
```
Comment (AdamKniec) [inline src/components/CountriesList/CountriesList.tsx]: switch to two separate functions and do not chain .filters()

File: src/components/CountriesList/CountriesList.tsx [hunk 3]
```diff
@@ -1,23 +1,31 @@
...
 			<StyledDisplaySection>
 				{!countries && "loading"}
-				{countries
-					?.filter((country) => {
-						return selectCountryRegion
-							? country.region === selectCountryRegion
-							: country;
-					})
-					.map(
+				{filteredData?.length ? (
```
Comment (AdamKniec) [inline src/components/CountriesList/CountriesList.tsx]: if(!filteredData) {
return empty state handler
} 

renderList

File: src/components/shared/input/Input.tsx [hunk 4]
```diff
@@ -9,9 +11,20 @@ export interface InputProps {
...
 
-export const Input = ({ inputId, placeholder }: InputProps) => (
-	<StyledInputLabel htmlFor={inputId}>
-		<SearchIcon />
-		<StyledInputElement id={inputId} placeholder={placeholder} />
-	</StyledInputLabel>
-);
+export const Input = ({ inputId, placeholder }: InputProps) => {
+	const { setInputValue } = useCountriesContext();
+	const inputHandle = (e: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (AdamKniec) [inline src/components/shared/input/Input.tsx]: -> handleInputChange

File: src/components/CountriesList/CountriesList.styles.tsx (hunk 1 above)
Comment (marcinel123) [reply src/components/CountriesList/CountriesList.styles.tsx]: changed.

File: src/components/CountriesList/CountriesList.tsx (hunk 3 above)
Comment (marcinel123) [reply src/components/CountriesList/CountriesList.tsx]: fixed.

File: src/components/shared/input/Input.tsx (hunk 4 above)
Comment (marcinel123) [reply src/components/shared/input/Input.tsx]: changed.

File: src/components/CountriesList/CountriesList.tsx (hunk 2 above)
Comment (marcinel123) [reply src/components/CountriesList/CountriesList.tsx]: fixed.

**Review by AdamKniec — APPROVED** [review]

**Pull request merged**